packages/commonwealth/server/migrations/20230807200113-rm-object-id.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.transaction(async (t) => {
+      // some new-thread-creation subscriptions are missing a chain_id
+      // creation of such subscriptions is no longer possible with the checks added below and the Sequelize model validation
+      await queryInterface.sequelize.query(
+        `
+        UPDATE "Subscriptions"
+        SET chain_id = object_id
+        WHERE category_id = 'new-thread-creation' AND chain_id IS NULL;
+      `,
+        { transaction: t }
+      );
+
+      await queryInterface.removeColumn('Subscriptions', 'object_id', {
+        transaction: t,
+      });
+
+      // https://www.postgresql.org/docs/current/sql-altertable.html#SQL-ALTERTABLE-NOTES
+      // The NOT VALID option means that PostgreSQL will not scan the entire table to ensure the constraint holds for
+      // existing records. This is okay since we know that all existing records are valid. This hugely increases
+      // performance since the checks can be committed directly.
+      await queryInterface.sequelize.query(
+        `
+        ALTER TABLE "Subscriptions"
+            ADD CONSTRAINT "chk_chain_id_on_chain_event"
+                CHECK ((category_id <> 'chain-event') OR (category_id = 'chain-event' AND chain_id IS NOT NULL)) NOT VALID,
+            ADD CONSTRAINT "chk_snapshot_id_on_snapshot_proposal"
```

Comment:
I worry about having undocumented constraints at the database level -- can these be reflected in the ORM? How will future developers know about these constraints?

Reply:
The exact same constraints are implemented as a validation function in the Sequelize model. Could add a wiki page describing how subscriptions work and in it include explanation of the constraints.

Reply:
@timolegros could you add an inline comment to the sequelize model explaining how the validation function is reflected at the DB level? So we can keep the docs together.